syntax_and_semantics/annotations.md

```diff
@@ -0,0 +1,176 @@
+# Annotations
+
+Some types and methods can be annotated with [built-in](./annotations/built_in_annotations.md) and user-defined annotations, such as the standard library's `JSON::Field` and `JSON::Options`.  
```

Comment:
"*Some*" doesn't make sense here. Is there any type or method that can't be annotated?
Also, the enumeration is already missing instance variables.

My proposal for an improved introduction:

> Annotations can be used to add metadata to certain features in the source code. Types, methods and instance variables may be annotated.
> User-defined annotations, such as the standard library's [`JSON::Field`](https://crystal-lang.org/api/JSON/Field.html), are defined using the `annotation` keyword. A number of [built-in annotations](./annotations/built_in_annotations.md) are provided by the compiler.